Use status.json insetad of status.ini

Luca0208 opened the pull request:
See #53
This currently has the following issues:
* It requires the user to manually go to the settings and reselect VATSIM as a network to update the URL to http://status.vatsim.net/status.json (Edit: Additionally this should probably use https right away)
* The server URL isn't extracted, because the new server file is also in .json format. I'm not sure whether the servers are ever actually queried from the server URL by QuteScoop, since the servers are also in the v3 data.
* Only one METAR and User status URL are extracted. In the status.ini format there was only one URL specified, however in the status.json there is an array. Unfortunately the documentation for the API seems to be down, so I can't check what the intended behaviour is.
* moveto0 and msg0 aren't extracted, since I don't know where in the JSON these are supposed to be. Again unfortunately the API documentation seems to be offline.

The last two points can be fixed once the API doc is back online, however I'd like your input on how to deal with the first two points.

Luca0208 committed:
Use status.json

jonaseberle commented:
>  It requires the user to manually go to the settings and reselect VATSIM as a network to update the URL to http://status.vatsim.net/status.json (Edit: Additionally this should probably use https right away)

Uargh I hate my old code ;) That toggle updates 2 settings values in `PreferencesDialog.cpp`. It would be nicer to just have that in `Settings::statusLocation` all the way.

Anyways I've added a settings migration step in https://github.com/qutescoop/qutescoop/pull/114 for cases where we might need to "migrate" settings - what do you think? 

And +1 for https:// all the way.

> The server URL isn't extracted, because the new server file is also in .json format. I'm not sure whether the servers are ever actually queried from the server URL by QuteScoop, since the servers are also in the v3 data.

You mean that `data.servers_all` in https://status.vatsim.net/status.json? Yes, I am pretty sure we haven't gotten server information from there.

> Only one METAR and User status URL are extracted. In the status.ini format there was only one URL specified, however in the status.json there is an array. Unfortunately the documentation for the API seems to be down, so I can't check what the intended behaviour is.

I think it's fine to just take the first. The possibility for multiple URLs is to balance load and the client is expected to choose an URL randomly, but in current times with VATSIM on Cloudflare I can't think of it being used ever.

> moveto0 and msg0 aren't extracted, since I don't know where in the JSON these are supposed to be. Again unfortunately the API documentation seems to be offline.

Let's keep it in mind but I don't think it is blocking now. We can add it later.

Luca0208 committed:
Use https instead of http

Luca0208 committed:
Remove download/statusLocation and add necessary migration

The previous approach stored download/statusLocation instead of
obtaining the necessary information from download/network.
The previous defaults were a user defined network, however the
statusLocation (Which is not the same as userLocation and never actually
displayed to the user) was already set to the VATSIM URL, causing
QuteScoop to display VATSIM data, but claim to use a user defined
network.

A migration has been added to change download/network to 0 (VATSIM) if
it was previously 1 (user defined) and the statusLocation matches the
old VATSIM status location

Luca0208 commented:
> Anyways I've added a settings migration step in https://github.com/qutescoop/qutescoop/pull/114 for cases where we might need to "migrate" settings - what do you think?

Looks good, I cherry picked that onto my branch. I've noticed while going through the settings code that the `statusLocation` is not actually derived from the `download/network` key which is what's causing the issue in the first place.
Also the defaults of these are weird, `download/network` defaults to 1 (user defined) while the standard `statusLocation` defaults to http://status.vatsim.net/ (For more details see  6e57454).
I've added some code to choose the appropriate status URL from the `download/network` key instead of having to make sure `download/network` and `statusLocation` are updated in lockstep.

I've also written an appropriate migration, changing `download/network` to 0 if it was previously 1 and the `statusLocation` was http://status.vatsim.net/ .
I'm also deleting `statusLocation` from the existing config, since it's unnecessary anyway.

This is probably ready to go.

jonaseberle commented:
Nice :) Thank you @Luca0208 !
